src/components/MapView/utils.ts

```diff
@@ -46,3 +49,83 @@ export const getFeatureInfoParams = (
 
   return params;
 };
+
+export const convertToTableData = (
+  result: ExposedPopulationResult,
+  groupBy: string,
+) => {
+  const { key, statistic } = result;
+  const { features } = result.featureCollection;
+  const fields = _.uniq(
+    _.map(features, f => f.properties && f.properties[key]),
+  );
+
+  const featureProperties = features.map(feature => {
+    return {
+      [groupBy]: feature.properties?.[groupBy],
+      [key]: feature.properties?.[key],
+      [statistic]: feature.properties?.[statistic],
+    };
+  });
+  const rowData = _.mapValues(_.groupBy(featureProperties, groupBy), x =>
+    _.chain(x)
+      .keyBy(key)
+      .mapValues(statistic)
+      .mapValues(z => _.parseInt(z))
+      .value(),
+  );
+
+  const groupedRowData = _.map(rowData, (x, i: number) => {
```

Comment:
try avoid using lodash methods if a vanilla approach would work just as well (reduces complexity)
Possibly every `map` used in this file could be replaced with the array.map

Reply:
Good point. I have updated the entire function to use vanilla functions as much as possible. Reduced complexity as well.